src/Illuminate/Support/Collection.php

```diff
@@ -939,6 +960,25 @@ public function transform(callable $callback)
     }
 
     /**
+     * Transpose each item in the collection, interchanging the row and column indexes.
+     *
+     * @return static
+     */
+    public function transpose()
+    {
+        $values = $this->values();
+
+        if (! $values->forAll('is_array')) {
+            throw new InvalidArgumentException('Can only transpose multi-dimensional arrays.');
+        }
```

Comment:
Another invariant that might be helpful to throw an exception against is making sure each child array is the same length. Ruby at least throws an exception in this case: http://ruby-doc.org/core-2.2.0/Array.html#method-i-transpose